corehq/apps/locations/adjacencylist.py

```diff
@@ -182,6 +179,12 @@ def make_cte_query(cte):
         query = (
             cte
             .join(self.all(), id=cte.col.id)
+            # EXISTS helps postgres avoid seq scan on locations table
+            # EXPLAIN ANALYZE showed postgres estimated > 1 million rows in the
+            # recursive CTE on softlayer when there were actually only 6 rows.
+            # The seq scan on the locations table took ~1 minute; EXISTS -> 3ms.
```

Comment:
:fearful: 
How do those compare on a large data set?

Reply:
EXISTS is optimized to stop looking when it finds the first matching row, so it should be fairly fast. I expect it will always be faster to optimize the query that way than to perform a seq scan on the locations table.